packages/engine-server/src/markdown/remark/utils.ts

```diff
@@ -216,6 +219,47 @@ export class LinkUtils {
   }
 }
 
+export class AnchorUtils {
+  static async findAnchors(opts: {
+    note: NoteProps;
+    wsRoot: string;
+  }): Promise<{ [index: string]: DNoteAnchorPositioned }> {
+    if (opts.note.stub) return {};
+    const noteContents = await NoteUtils.readFullNote(opts);
+    const noteAnchors = RemarkUtils.findAnchors(noteContents);
+    const anchors: [string, DNoteAnchorPositioned][] = [];
+    noteAnchors.forEach((anchor) => {
+      if (_.isUndefined(anchor.position)) return;
+      const { line, column } = anchor.position.start;
+      if (anchor.type === "heading") {
```

Comment:
to be consistent, we might just add this as a `DendronASTTypes` of heading

Reply:
Added `HEADING` to `DendronASTTypes` and revised existing code to use it.